MaterialSkin/HTML/material/html/js/queue-page.js

```diff
@@ -69,8 +71,10 @@ function buildArtistAlbumLines(i, queueAlbumStyle, queueContext) {
         artistAlbumContext = undefined;
     }
     if (!queueAlbumStyle || !artistIsRemoteTitle) {
-        artistAlbum = addPart(artistAlbum, buildWorkLine(i, artistStr, 'queue'));
```

Comment:
You have removed the call to `buildWorkLine` (bad name, as its not a line, but...). This is used to make the work part of the string clickable links (if enabled). This function is in `itemlinks.js`. For consistency you need to use this function, but feel free to adjust it as required (its only used here).

Reply:
Ah yes, I stopped using that because I couldn't work out how to use `addPart` to put the "added part" at the beginning of the resulting string. I meant to go back to that once I got things looking the way I wanted but forgot.

What do you think about a 2 line heading in certain circumstances? Ideally only when dealing with a Work, not when simply a Grouping (like the Stranglers example)? Is it even possible? (I did try putting a `<br />` in the string, but that resulted in the first detail line being rendered on top of the second heading line.

If the 2 line option for Works is feasible, that would allow room for the standard Artist, Band, Conductor combination instead of just artist for classical works.

Reply:
Not keen on a second line - the whole idea of this 'view' was to be compact. I'm not sure why you insist on having so much info on display. The first screenshot, to my mind, has way too much text. But that's just my opinion.

'addPart' always appends - this function is in `utils.js` Just changing the order should suffice, e.g. (not tested):

```javascript
artistAlbum = addPart(buildWorkLine(i, artistStr, 'queue'), artistAlbum);
```

Reply:
The reason is that it is `Composer * Work * Grouping` which describes the group. 

I agree the first example looks messy, and it's not helped by there being only one track in the first group, which would almost always be the case for operas as they start with an overture. 2 lines might actually help here, obviously we don't want to lose the album link (though we could possibly move artist into the detail line - the one used in the header can be a bit random - changes with rescans - when there's more than one) and include in the detail line.

I would envisage the 2 lines only kicking in if we've got a work _and_ a grouping, so it's under the control of the user's tagging: my thinking is that if they've gone to the trouble of adding a Grouping tag, it's because they want to see it.

I'm guessing that to implement this the template would need to change around here:
https://github.com/CDrummond/lms-material/blob/e2bf13d5beddf50b7e59144a182df6dc19addb41/MaterialSkin/HTML/material/html/js/queue-page.js#L253

Reply:
For 2 lines you also need to set the size of each item - see the "size:" setting on line 169

However, if multi-lines are required then perhaps it's just not suitable to this view style?

Reply:
I think for "tag heavy" stuff like operas, the hierarchical view is better suited than repeating everything on every track line. But let me throw something together and then you can see what you think.

Reply:
I've opened up space for the track title:

![image](https://github.com/user-attachments/assets/ab392e42-67b1-4964-a1c4-1b45cb50ffea)

but have spent hours messing with the template but can only seem to push the text down into the gap by changing the stylesheet (and the track time does not go with it):

![image](https://github.com/user-attachments/assets/fc5fdf19-d50e-4965-a7c5-28e0902a0e47)

What am I missing?

Reply:
> Also, in the above, why repeat "Violin Concerto in D Minor, Op. 47" on each line?

That's my track name tag. In my test database I have a mixture of it set to WORK:MOVEMENT  and just MOVEMENT, so I can test the variations that I know people use out there.

Reply:
Sorry, that I'm not more helpful at the moment - I'm busy at work and at home at the moment! If you work on splitting the text into what's required I'll look at the CSS / template changes.